### PR TITLE
bugfix/6657-contrast-column-datalabels

### LINKED ISSUE #6657: Wrong color of first and last datalabel [5.0.10]
#### Expected behaviour
Display all datalabels in dark colour.
screen shot 2017-04-28 at 15 36 50

#### Actual behaviour
First and last datalabels are coulored with white
screen shot 2017-04-28 at 15 37 40

#### Live demo with steps to reproduce
https://jsfiddle.net/c1rcxn0r/ [5.0.10] - Issue
https://jsfiddle.net/c1rcxn0r/1 [5.0.9] - Works OK

### PR DESCRIPTION
**Pull request by pawelfus**
Fixed #6657, inappropriate contrast color for first and last dataLabel in column/bar series.
___
Simply moved `intersectRect` to Utils, I think this function is quite handy.
Let me know if TS definitions should look different (personally, I think it's more readable this way).

### CODE REVIEW

File: js/parts/Utilities.js [hunk 1]
```diff
@@ -1354,6 +1354,37 @@ H.format = function (str, ctx, time) {
 H.getMagnitude = function (num) {
     return Math.pow(10, Math.floor(Math.log(num) / Math.LN10));
 };
+/**
+ * Get the magnitude of a number.
```
Comment (TorsteinHonsi) [inline js/parts/Utilities.js]: Description is non-descriptive

File: js/parts/Utilities.js [hunk 2]
```diff
@@ -1354,6 +1354,37 @@ H.format = function (str, ctx, time) {
...
+ *        x-position of the second box
+ * @param {number} y2
+ *        y-position of the second box
+ * @param {number} w2
+ *        width of the second box
+ * @param {number} h2
+ *        height of the second box
+ *
+ * @return {boolean}
+ *         Boolean wheater rects are intersecting.
```
Comment (TorsteinHonsi) [inline js/parts/Utilities.js]: Typo: whether

File: ts/parts/DataLabels.ts [hunk 3]
```diff
@@ -2180,7 +2181,20 @@ if (seriesTypes.column) {
         );
 
         // If label was justified and we have contrast, set it:
-        if (point.isLabelJustified && point.contrastColor) {
+        if (
+            point.contrastColor &&
+            point.shapeArgs &&
+            intersectRect(
```
Comment (TorsteinHonsi) [inline ts/parts/DataLabels.ts]: It's probably better to refactor the function to take two objects, each having `{ x, y, width, height }`, so we don't have to pass over eight nested parameters.

Comment (bre1470) [reply ts/parts/DataLabels.ts]: Then it is actually better to have two parameter, because we have interfaces like `PositionObject` and `DimensionObject`.

Comment (bre1470) [reply ts/parts/DataLabels.ts]: Even better, we have the `BBoxObject` interface, that takes all four properties.

**Pull request merged**